evm_loader/program/src/executor_state.rs

```diff
@@ -437,17 +437,20 @@ impl ExecutorSubstate {
         transfer: &Transfer,
         backend: &B,
     ) -> Result<(), ExitError> {
+        let min_decimals = u32::from(token::eth_decimals() - token::token_mint::decimals());
+        let min_value = U256::from(10_u64.pow(min_decimals));
+        let transfer_value_without_min_value = transfer.value - transfer.value % min_value;
         {
             let source = self.account_mut(transfer.source, backend);
             if source.basic.balance < transfer.value {
                 return Err(ExitError::OutOfFund);
             }
-            source.basic.balance -= transfer.value;
+            source.basic.balance -= transfer_value_without_min_value;
```

Comment:
Shouldn't we compare `source.basic.balance` with `transfer_value_without_min_value` as we only sub this one.
It would be simpler and clearer to use `checked_sub` here:
```rust
 source.basic.balance
     .checked_sub(transfer_value_without_min_value)
     .ok_or(ExitError::OutOfFund)?;
```

Reply:
Fixed